packages/monitor-v2/src/bot-oo-v3/SettleAssertions.ts

```diff
@@ -0,0 +1,43 @@
+import { getContractInstanceWithProvider, Logger, MonitoringParams, OptimisticOracleV3Ethers } from "./common";
+import { logSettleAssertion } from "./BotLogger";
+
+export async function settleAssertions(logger: typeof Logger, params: MonitoringParams): Promise<void> {
+  const oo = await getContractInstanceWithProvider<OptimisticOracleV3Ethers>("OptimisticOracleV3", params.provider);
+
+  const assertions = await oo.queryFilter(oo.filters.AssertionMade(), undefined, undefined);
+  const assertionsSettled = await oo.queryFilter(oo.filters.AssertionSettled(), undefined, undefined);
+
+  const assertionsToSettle = assertions.filter(
+    (assertion) => !assertionsSettled.some((settled) => settled.args.assertionId === assertion.args.assertionId)
+  );
```

Comment:
I might suggest as a minor optimization: adding the settled assertion ids to a Set and then checking for membership in the set in this filter. It's not a problem at low numbers of assertions, but it's simple and it future proofs this for when there are many assertions.